packages/e2e-tests/specs/editor/various/reusable-blocks.test.js

```diff
@@ -272,4 +273,40 @@ describe( 'Reusable blocks', () => {
 
 		expect( console ).not.toHaveErrored();
 	} );
+
+	it( 'should be able to insert a reusable block twice', async () => {
+		await createReusableBlock(
+			'Awesome Paragraph',
+			'Random reusable block'
+		);
+		await saveAll();
+		await clearAllBlocks();
+		await insertReusableBlock( 'Random reusable block' );
+		await insertReusableBlock( 'Random reusable block' );
+		await saveDraft();
+		await page.reload();
+
+		// Replace the content of the  first paragraph
+		const paragraphBlock = await page.waitForSelector(
+			'.block-editor-block-list__block[data-type="core/paragraph"]'
+		);
+		paragraphBlock.focus();
+		await pressKeyWithModifier( 'primary', 'a' );
+		await page.keyboard.press( 'ArrowRight' );
+		await page.keyboard.type( '*' );
```

Comment:
If I wrote more than a single char, I notice that focus loss can happen in the test. I wasn't able to reproduce the issue while testing manually but we probably have another hidden bug.